container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/JerseyServletProvider.java

```diff
@@ -67,9 +67,8 @@ private ResourceConfig resourceConfig(RestApiContext restApiContext) {
             ResourceOrProviderClassVisitor visitor = ResourceOrProviderClassVisitor.visit(new ClassReader(inputStream));
             return Optional.ofNullable(visitor.getClassName());
         } catch (IOException e) {
-            // ignored
+            throw new RuntimeException(e);
         }
-        return Optional.empty();
```

Comment:
The previous version here as questioned by @bjorncs was slightly different in behaviour than the Scala version. This change should bring it closer.